python/setup.py

```diff
@@ -71,7 +100,17 @@ def _build_with_cmake(self, ext: CMakeExtension) -> None:
             ["cmake", "--build", ".", *build_args], cwd=build_temp, check=True
         )
 
+
+def parse(root, config):
```

Comment:
Original parse also has `**kvargs` parameters: https://github.com/pypa/setuptools_scm/blob/d75201a03ea01bf2d99a2521b87cdea30320f887/src/setuptools_scm/git.py#L200-L205 which should be passed through.

Reply:
Actually function should implement the following protocol: https://github.com/pypa/setuptools_scm/blob/d75201a03ea01bf2d99a2521b87cdea30320f887/src/setuptools_scm/_config.py#L54-L57

Reply:
So `def parse(root, *, config):` ?

Reply:
Could you please check what this weird syntax does mean in Python and write a proper signature which passes all the parameters if you are still going to override `parse`.